### PR TITLE
Use genreconciler framework in the eventing reconciler.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Proposed Changes

The genreconciler framework takes a lot of the boilerplate away, so there's no reason not to use it, is there?

**Release Note**

```release-note
NONE
```

/assign @aliok @houshengbo

### CODE REVIEW

**Comment by markusthoemmes** [comment]
WIPing this, I want to do a different change first.

**Comment by markusthoemmes** [comment]
Memo to self: Make sure the old finalizer gets removed from existing resources.

**Comment by houshengbo** [comment]
@markusthoemmes I think this one should go in first as the first step of refactoring.

File: pkg/reconciler/knativeeventing/controller.go [hunk 1]
```diff
@@ -53,31 +54,31 @@ func NewController(
...
-	impl := controller.NewImpl(c, c.Logger, reconcilerName)
+	c := &Reconciler{
+		kubeClientSet: kubeclient.Get(ctx),
+		config:        config,
+		eventings:     sets.NewString(),
+	}
+	impl := knereconciler.NewImpl(ctx, c)
 
-	c.Logger.Info("Setting up event handlers for %s", reconcilerName)
+	logger.Info("Setting up event handlers")
```
Comment (houshengbo) [inline pkg/reconciler/knativeeventing/controller.go:81]: Do we need to the keep it the same as before by incuding reconcilerName? Or do we have other reasons to remove it?

**Comment by markusthoemmes** [comment]
@houshengbo WFM, I'm indifferent to merging order. I'll have to rebase everything anyway.

File: pkg/reconciler/knativeeventing/controller.go [hunk 2]
```diff
@@ -53,31 +54,31 @@ func NewController(
...
 	}
 
-	config, err := mfc.NewManifest(filepath.Join(koDataDir, "knative-eventing/"), cfg,
-		mf.UseLogger(zapr.NewLogger(c.Logger.Desugar()).WithName("manifestival")))
+	config, err := mfc.NewManifest(filepath.Join(koDataDir, "knative-eventing/"),
+		cfg,
+		mf.UseLogger(zapr.NewLogger(logger.Desugar()).WithName("manifestival")))
 	if err != nil {
-		c.Logger.Error(err, "Error creating the Manifest for knative-eventing")
+		logger.Error(err, "Error creating the Manifest for knative-serving")
```
Comment (houshengbo) [inline pkg/reconciler/knativeeventing/controller.go:70]: Should be `knative-eventing` here.

File: pkg/reconciler/knativeeventing/knativeeventing.go [hunk 3]
```diff
@@ -50,121 +50,124 @@ var (
 	rolebinding mf.Predicate = mf.Any(mf.ByKind("ClusterRoleBinding"), mf.ByKind("RoleBinding"))
 )
 
-// Reconciler implements controller.Reconciler for Knativeeventing resources.
+// Reconciler implements controller.Reconciler for Knativeserving resources.
```
Comment (houshengbo) [inline pkg/reconciler/knativeeventing/knativeeventing.go:53]: s/Knativeserving/Knativeeventing

File: pkg/reconciler/knativeeventing/knativeeventing.go [hunk 4]
```diff
@@ -50,121 +50,124 @@ var (
...
-	if apierrs.IsNotFound(err) {
-		return nil
-	} else if err != nil {
-		r.Logger.Error(err, "Error getting KnativeEventings")
-		return err
-	}
+var _ knereconciler.Interface = (*Reconciler)(nil)
+var _ knereconciler.Finalizer = (*Reconciler)(nil)
+
+// FinalizeKind removes all resources after deletion of a KnativeServing.
```
Comment (houshengbo) [inline pkg/reconciler/knativeeventing/knativeeventing.go:67]: s/KnativeServing/KnativeEventing

File: pkg/reconciler/knativeeventing/controller.go (hunk 1 above)
Comment (markusthoemmes) [reply pkg/reconciler/knativeeventing/controller.go:81]: Good point, I shall remove the globals. I don't think there's any value in these anyway. The logger will be setup to have the right context straight away.

Comment (markusthoemmes) [reply pkg/reconciler/knativeeventing/controller.go:81]: Logs look like this:

```
{"level":"info","ts":"2020-04-27T15:49:54.622Z","logger":"eventing-operator","caller":"knativeeventing/controller.go:76","msg":"Setting up event handlers"}
```

There's enough context there imo.

**Comment by markusthoemmes** [comment]
@houshengbo thanks for the review, fixes pushed.

**Pull request merged**